plugin-hrm-form/src/components/case/documentUploadHandler.ts

```diff
@@ -43,28 +44,56 @@ const uploadDocument = async (file: File, preSignedUrl: string, mimeType: string
  *
  * It returns the file name at AWS
  */
-const onFileChange = async event => {
+const bindOnFileChange = (caseId: number) => async event => {
   const file = event.target.files[0];
+  const { name, size, type } = file;
 
-  if (file.size > MAX_FILE_SIZE) {
+  if (size > MAX_FILE_SIZE) {
     alert('File exceeds max size.');
     return '';
   }
+  const mimeType = type;
+  const { docsBucket: bucket } = getHrmConfig();
 
-  const mimeType = file.type;
-  const response = await getFileUploadUrl(file.name, mimeType);
-  await uploadDocument(file, response.uploadUrl, mimeType);
+  const key = `${new Date().getTime()}-${name}`;
 
-  return response.fileNameAtAws;
-};
+  const { media_url: preSignedUrl } = await fetchHrmApi(
+    generateSignedURLPath({
+      method: 'putObject',
+      objectType: 'case',
+      objectId: caseId.toString(),
+      fileType: 'document',
+      location: {
+        bucket,
+        key,
+      },
+    }),
+  );
+
+  await uploadDocument(file, preSignedUrl, mimeType);
 
-const onDeleteFile = async (fileName: string) => {
-  await deleteFile(fileName);
+  return key;
 };
 
-const fileUploadCustomHandlers: CustomHandlers = {
-  onFileChange,
-  onDeleteFile,
+const bindOnDeleteFile = (caseId: number) => async (fileName: string) => {
+  const { docsBucket: bucket } = getHrmConfig();
+  await fetchHrmApi(
+    generateSignedURLPath({
+      method: 'deleteObject',
+      objectType: 'case',
+      objectId: caseId.toString(),
+      fileType: 'document',
+      location: {
+        bucket,
+        key: fileName,
+      },
+    }),
+  );
 };
 
-export default fileUploadCustomHandlers;
+export const bindFileUploadCustomHandlers = (caseId: number) => (): CustomHandlers => {
+  return {
+    onFileChange: bindOnFileChange(caseId),
+    onDeleteFile: bindOnDeleteFile(caseId),
+  };
+};
```

Comment:
Extremely small note: 
You can 
```suggestion
export const bindFileUploadCustomHandlers = (caseId: number): CustomHandlers => {
  return {
    onFileChange: bindOnFileChange(caseId),
    onDeleteFile: bindOnDeleteFile(caseId),
  };
};
```
then when you call it you can remove one set of parenthesis.